src/main/java/com/synopsys/integration/alert/web/api/upload/UploadEndpointController.java

```diff
@@ -24,54 +24,61 @@
 
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.multipart.MultipartFile;
 
 import com.synopsys.integration.alert.common.action.UploadEndpointManager;
 import com.synopsys.integration.alert.common.rest.ResponseFactory;
+import com.synopsys.integration.alert.common.rest.model.ExistenceModel;
 
 @RestController
 @RequestMapping(UploadEndpointManager.UPLOAD_ENDPOINT_URL)
 public class UploadEndpointController {
-
     public static final String TARGET_KEY_MISSING = "Must be given the key associated with the custom functionality.";
+
     private final UploadEndpointManager uploadEndpointManager;
-    private final ResponseFactory responseFactory;
 
     @Autowired
-    public UploadEndpointController(UploadEndpointManager uploadEndpointManager, ResponseFactory responseFactory) {
+    public UploadEndpointController(UploadEndpointManager uploadEndpointManager) {
         this.uploadEndpointManager = uploadEndpointManager;
-        this.responseFactory = responseFactory;
     }
 
     @GetMapping("/{key}/exists")
-    public ResponseEntity<String> checkUploadedFileExists(@PathVariable String key) {
+    public ExistenceModel checkUploadedFileExists(@PathVariable String key) {
         if (StringUtils.isBlank(key)) {
-            return responseFactory.createBadRequestResponse("", TARGET_KEY_MISSING);
+            throwTargetKeyMissingException();
         }
```

Comment:
This class also has a number of duplicate if statements that can be put into the private method.